RCNN_tf.py

```diff
@@ -0,0 +1,338 @@
+import sys
+import tensorflow as tf
+import numpy as np
+from load_data import load_mnist, preprocess_mnist_data
+
+# Constants
+# float64 is not allowed by all tf operations
+PRECISION_TF = tf.float32
+PRECISION_NP = np.float32
+PADDING_LIST = ['SAME', 'VALID']
+
+
+def rcl(
+        input_data,
+        num_input_channels,
+        num_filter,
+        filter_shape,
+        depth=3,
+        std=.03,
+        alpha=1e-3,
+        beta=.75,
+        normalization_feature_maps=8,
+        name='rcl'
+):
+    conv_filter_forward_shape = [filter_shape[0], filter_shape[1], num_input_channels, num_filter]
+    conv_filter_recurrent_shape = [filter_shape[0], filter_shape[1], num_filter, num_filter]
+    recurrent_cells_shape = [1, input_data.shape[1], input_data.shape[2], num_filter]
+
+    cell_states = tf.Variable(
+        tf.zeros(recurrent_cells_shape, dtype=PRECISION_TF),
+        trainable=False,
+        name=name + '_cells'
+    )
+
+    forward_weights = tf.Variable(
+        tf.truncated_normal(conv_filter_forward_shape, stddev=std, dtype=PRECISION_TF),
+        trainable=True,
+        name=name + '_forward'
+    )
+
+    recurrent_weights = tf.Variable(
+        tf.truncated_normal(conv_filter_recurrent_shape, stddev=std, dtype=PRECISION_TF),
+        trainable=True,
+        name=name + '_recurrent'
+    )
+
+    bias = tf.Variable(
+        tf.truncated_normal([num_filter], dtype=PRECISION_TF),
+        trainable=True,
+        name=name + '_bias'
+    )
+
+    forward_output = tf.nn.conv2d(input_data, forward_weights, [1, 1, 1, 1], padding='SAME')
+
+    output = None
+    for _ in range(depth):
+        recurrent_output = tf.nn.conv2d(cell_states, recurrent_weights, [1, 1, 1, 1], padding='SAME')
+
+        output = tf.add(forward_output, recurrent_output)
+        output += bias
+
+        cell_states = tf.nn.local_response_normalization(
+            tf.maximum(np.asarray(0.0).astype(PRECISION_NP), output),
+            depth_radius=normalization_feature_maps,
+            bias=1,
+            alpha=alpha,
+            beta=beta,
+            name=name + '_lrn'
+        )
+
+    return output
+
+
+def convolutional_layer(
+        input_data,
+        filter_shape,
+        num_input_channels,
+        num_filter,
+        std=.03,
+        stride=(1, 1),
+        padding='same',
+        name='conv'
+):
+    conv_filter_shape = [filter_shape[0], filter_shape[1], num_input_channels, num_filter]
+    strides = [1, stride[0], stride[1], 1]
+
+    weights = tf.Variable(
+        tf.truncated_normal(conv_filter_shape, stddev=std, dtype=PRECISION_TF),
+        trainable=True,
+        name=name + '_weights'
+    )
+
+    if padding.upper() not in PADDING_LIST:
+        raise ValueError('Padding value is not understood')
+
+    output = tf.nn.conv2d(input_data, weights, strides=strides, padding=padding.upper())
+
+    return output
+
+
+def pooling_layer(
```

Comment:
I get that you want to use 'pure' tf, but i feel like this introduces more potential for error. what is wrong with using [tf.nn.pool](https://www.tensorflow.org/api_docs/python/tf/nn/pool) ?

Reply:
Take a look some lines below :). This is just to provide a simple interface